### ISSUE TITLE
Add Word Count

### ISSUE THREAD

**Issue by bojjenclon** [fabiospampinato/notable#196]
Many text editors have a status bar underneath the editable area with info such as the current line, current column, and total word count. I think it would be beneficial to have all of this, but at the very least word count would be a nice feature to have.

**Comment by bojjenclon** [comment]
Note: if interested, I may be willing to help with a PR depending on how you'd prefer to go about implementing this.

**Comment by fabiospampinato** [comment]
We are probably not going to add a statusbar, but word count could be added as part of some extra metadata #67.

Regarding the implementation I was thinking about maybe adding button to the mainbar that trigger a popover, I'm not sure if that would clutter the design too much though. If you want to submit a PR for this I'd be happy to consider it.

**Comment by bojjenclon** [comment]
Depending on the decision made here, I may end up forking this with a status bar for personal use. I already have a basic implementation up locally with a word count widget (although I can't say for certain I'm properly adhering to the project structure, as I didn't study it long before whipping up the status bar).

**Comment by bojjenclon** [comment]
Proof of concept

![notable proof of concept](https://user-images.githubusercontent.com/1127397/50716727-f1b35280-1050-11e9-8bdf-29907f3c0403.gif)

**Comment by fabiospampinato** [comment]
> >We are probably not going to add a statusbar
>
> Sorry I have not been closely keeping up with all of these issues, but is there a reason for this? I could see line length fitting there as well for those who like to keep a boundary

@smvoss I don't want to sacrifice esthetics for this. Word count can be added to a separate popover that's just a shortcut away.

**Comment by smvoss** [comment]
Thats too bad, I think @bojjenclon has a really good mockup.

Maybe just have it toggleable, off by default?

**Comment by fabiospampinato** [comment]
@smvoss I don't think it looks "really good", I see a huge amount of white space, and the bigger the screen the more wasted space there is. Maybe this could be supported via plugins #128, or there's a setting for this in CodeMirror that you'll be able to enable when they become customizable.

**Comment by smvoss** [comment]
>  I see a huge amount of white space, and the bigger the screen the more wasted space there is. 

Realistically, the size of the screen would improve the usable space to wasted space ratio, not waste any more space.

**Comment by bojjenclon** [comment]
Personally, I hate having to go through extra menus for something like word count, as I consider it a key part of writing. I use applications like this for more than just "notes," but also for creative writing organization. So word count is something I like to have at a glance.

Plus I feel like with this being an Electron app, the odds of it being run on low end systems is relatively low. So I would expect most of the user base to have at least a 1080p monitor.

But if its supportable via plugin, I certainly think that'd be a viable approach that could keep all sides happy.

**Comment by fabiospampinato** [comment]
> Realistically, the size of the screen would improve the usable space to wasted space ratio, not waste any more space.

The 2 are not really mutually exclusive, the usable-to-wasted ratio increases but the wasted space increases also, plus I don't like it from an aesthetic perspective.

But I do recognize that especially in the future where we have plugins having a statusbar could be useful, also for plugins as a place where they could add their stuff.

**Comment by bojjenclon** [comment]
Is there a reason against making it toggleable? I understand you not liking it aesthetically, but a lot of people do. So it seems to me a smart choice could be to have a toggleable status bar, so future plugins could utilize it, and anyone who doesn't want it can just turn it off.

**Comment by fabiospampinato** [comment]
> Is there a reason against making it toggleable? 

None, feel free to submit a PR implementing this.

**Comment by fabiospampinato** [comment]
Let's add a statusbar first #718
